From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sun, 26 Jun 2022 11:08:40 +0200
Subject: [PATCH] Include BungeeGuard

Automatically downloads BungeeGuard plugin if bungeeguard support is enabled in spigot.yml

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index dc74094062e1347c428abcdac1c0276a3690dafe..4df1fcbfc2b47b630aecee222ed99217899a2fb8 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -411,6 +411,7 @@ public final class CraftServer implements Server {
 
         File pluginFolder = this.getPluginsFolder(); // Paper
         boolean sparkLoaded = false; // Purpur
+        boolean bungeeGuardLoaded = false; // Sharkur - Include BungeeGuard
 
         // Paper start
         if (true || pluginFolder.exists()) {
@@ -425,6 +426,7 @@ public final class CraftServer implements Server {
                     plugin.getLogger().info(message);
                     plugin.onLoad();
                     sparkLoaded = sparkLoaded || plugin.getDescription().getName().equalsIgnoreCase("spark"); // Purpur
+                    bungeeGuardLoaded = bungeeGuardLoaded || plugin.getDescription().getName().equalsIgnoreCase("BungeeGuard"); // Sharkur - Include BungeeGuard
                 } catch (Throwable ex) {
                     Logger.getLogger(CraftServer.class.getName()).log(Level.SEVERE, ex.getMessage() + " initializing " + plugin.getDescription().getFullName() + " (Is it up to date?)", ex);
                 }
@@ -457,6 +459,35 @@ public final class CraftServer implements Server {
             }
         }
         // Purpur end
+
+        // Sharkur start - Include BungeeGuard
+        if (!bungeeGuardLoaded && org.spigotmc.SpigotConfig.bungee) {
+            org.sharkurmc.sharkur.SharkurLogger.LOGGER.warning("You have bungeecord support enabled but you don't have BungeeGuard plugin!");
+            org.sharkurmc.sharkur.SharkurLogger.LOGGER.warning("Your server may simply be griefed if you have online mode turned off.!");
+            org.sharkurmc.sharkur.SharkurLogger.LOGGER.warning("Sharkur will automatically download BungeeGuard. After downloading, shut down the server and install it on your proxy.");
+            org.sharkurmc.sharkur.SharkurLogger.LOGGER.warning("Then configure it in plugins/BungeeGuard/config.yml");
+            try {
+                File file = new File("cache", "BungeeGuard.jar");
+                try (java.io.BufferedInputStream in = new java.io.BufferedInputStream(new java.net.URL("https://github.com/lucko/BungeeGuard/releases/latest/download/BungeeGuard.jar").openStream()); java.io.FileOutputStream out = new java.io.FileOutputStream(file)) {
+                    byte[] dataBuffer = new byte[1024];
+                    int bytesRead;
+                    while ((bytesRead = in.read(dataBuffer, 0, 1024)) != -1) {
+                        out.write(dataBuffer, 0, bytesRead);
+                    }
+                    out.flush();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                Plugin bungeeGuard = this.pluginManager.loadPlugin(file);
+                String message = String.format("Loading %s", bungeeGuard.getDescription().getFullName());
+                bungeeGuard.getLogger().info(message);
+                bungeeGuard.onLoad();
+            } catch (Exception e) {
+                getLogger().severe("Failed to download and load BungeeGuard plugin");
+                e.printStackTrace();
+            }
+        }
+        // Sharkur end
     }
 
     // Paper start
